tests/test.rs

```diff
@@ -431,6 +431,22 @@ test!(test_getset_plain_password, {
     assert!(sock.get_mechanism().unwrap() == Mechanism::ZMQ_NULL);
 });
 
+test!(test_zmq_set_xpub_verbose, {
+    let ctx = Context::new();
+    let xpub = ctx.socket(XPUB).unwrap();
+    let sub = ctx.socket(SUB).unwrap();
+
+    xpub.bind("inproc://set_xpub_verbose").unwrap();
+    xpub.set_xpub_verbose(1).unwrap();
+
+    sub.connect("inproc://set_xpub_verbose").unwrap();
+    sub.set_subscribe("topic".as_bytes()).unwrap();
```

Comment:
I'm not sure the test actually tests if `set_xpub_verbose()` worked, from what I gather from the `zmq_setsockopt` manpage on ZMQ_XPBUB_VERBOSE`, the first subscribe message for a topic is always passed to the XPUB socket; I think you should try to subscribe twice, and then see if you actually get two messages.

Reply:
You are right this is not a proper test.